src/SFA.DAS.Roatp.CourseManagement.Jobs/Infrastructure/ApiClients/ApiClientBase.cs

```diff
@@ -0,0 +1,153 @@
+        /// <exception cref="HttpRequestException">Thrown if something unexpected occurred when sending the request.</exception>
+        protected async Task<T> Get<T>(string uri)
+        {
+            try
+            {
+                using (var response = await _httpClient.GetAsync(new Uri(uri, UriKind.Relative)))
+                {
+                    await LogErrorIfUnsuccessfulResponse(response);
+                    return await response.Content.ReadAsAsync<T>();
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, $"Error when processing request: {HttpMethod.Get} - {uri}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// HTTP POST to the specified URI
+        /// </summary>
+        /// <param name="uri">The URI to the end point you wish to interact with.</param>
+        /// <returns>The HttpStatusCode, which is the responsibility of the caller to handle.</returns>
+        /// <exception cref="HttpRequestException">Thrown if something unexpected occurred when sending the request.</exception>
+        protected async Task<HttpStatusCode> Post<T>(string uri, T model)
+        {
+            var serializeObject = JsonConvert.SerializeObject(model);
+
+            try
+            {
+                using (var response = await _httpClient.PostAsync(new Uri(uri, UriKind.Relative),
+                    new StringContent(serializeObject, Encoding.UTF8, _contentType)))
+                {
+                    await LogErrorIfUnsuccessfulResponse(response);
+                    return response.StatusCode;
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, $"Error when processing request: {HttpMethod.Post} - {uri}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// HTTP POST to the specified URI
+        /// </summary>
+        /// <typeparam name="T">The type of the object to POST.</typeparam>
+        /// <typeparam name="U">The type of the object to read.</typeparam>
+        /// <param name="uri">The URI to the end point you wish to interact with.</param>
+        /// <returns>A Task yielding the result (of type U).</returns>
+        /// <exception cref="HttpRequestException">Thrown if something unexpected occurred when sending the request.</exception>
+        protected async Task<U> Post<T, U>(string uri, T model)
+        {
+            var serializeObject = JsonConvert.SerializeObject(model);
+
+            try
+            {
+                using (var response = await _httpClient.PostAsync(new Uri(uri, UriKind.Relative),
+                    new StringContent(serializeObject, Encoding.UTF8, _contentType)))
+                {
+                    await LogErrorIfUnsuccessfulResponse(response);
+                    return await response.Content.ReadAsAsync<U>();
```

Comment:
Same here, potentially this will error on unsuccessful response

Reply:
again, try catch deals with it?

Reply:
now added a response.IsSuccessfulStatusCode to deal with this